src/layout/umap.c

```diff
@@ -91,28 +149,92 @@ static igraph_error_t igraph_i_umap_find_sigma(const igraph_vector_t *distances,
     return IGRAPH_SUCCESS;
 }
 
+/* Check the distances argument, which should be NULL or a vector of nonnegative numbers */
+static igraph_error_t igraph_i_umap_check_distances(const igraph_vector_t *distances, igraph_integer_t no_of_edges) {
+
+    if (distances == NULL) {
+        return IGRAPH_SUCCESS;
+    }
+
+    if (igraph_vector_size(distances) != no_of_edges) {
+        IGRAPH_ERROR("Distances must be the same number as the edges in the graph.", IGRAPH_EINVAL);
+    }
+
+    for (igraph_integer_t eid = 0; eid != no_of_edges; eid++) {
+        if (VECTOR(*distances)[eid] < 0) {
+            IGRAPH_ERROR("Distances cannot be negative.", IGRAPH_EINVAL);
+        } else if (isnan(VECTOR(*distances)[eid])) {
+            IGRAPH_ERROR("Distances cannot contain NaN values.", IGRAPH_EINVAL);
+        }
+    }
+
+    return IGRAPH_SUCCESS;
+}
 
-/* Convert the graph with distances into a probability graph with exponential decay */
-/* NOTE: this is funny because the distance was a correlation to start with... ?? */
-static igraph_error_t igraph_i_umap_find_prob_graph(const igraph_t *graph,
-        const igraph_vector_t *distances, igraph_vector_t *umap_weights) {
+
+/**
+ * \function igraph_layout_umap_compute_connectivities
+ * \brief Compute connectivities (edge weights) for a UMAP layout starting from distances.
+ *
+ * \experimental
+ *
+ * UMAP is used to embed high-dimensional vectors in a low-dimensional space
+ * (most commonly 2D). It uses a distance graph as an intermediate data structure,
+ * making it also a useful graph layout algorithm. See igraph_layout_umap for more information.
```

Comment:
```suggestion
 * making it also a useful graph layout algorithm. See \ref igraph_layout_umap() for more information.
```

Use `\ref` to refer to functions. Add `()` after function names to indicate functions clearly.